lib/pure/asyncdispatch.nim

```diff
@@ -167,8 +167,13 @@ type
     callbacks: Deque[proc ()]
 
 proc processTimers(p: PDispatcherBase) {.inline.} =
+  #Process just part if timers at a step
+  var part = newSeq[tuple[finishAt: float, fut: Future[void]]]()
```

Comment:
Most of the time no timers are involved, so please do the sequence allocation lazily. (Start with ``nil``.)